controllers/csiaddonsnode_controller.go

```diff
@@ -39,17 +50,98 @@ type CSIAddonsNodeReconciler struct {
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the CSIAddonsNode object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
 func (r *CSIAddonsNodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx)
+
+	logger.Info("Starting Reconcile")
+	// Fetch CSIAddonsNode instance
+	instance := &csiaddonsv1alpha1.CSIAddonsNode{}
+	err := r.Client.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			logger.Info("CSIAddonsNode resource not found")
+
+			return ctrl.Result{}, nil
+		}
+
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	nodeID := instance.Spec.Driver.NodeID
+	driverName := instance.Spec.Driver.Name
+	endPoint := instance.Spec.Driver.EndPoint
+	key := r.creatKey(instance.Namespace, instance.Name)
+
+	logger = logger.WithValues(
+		"NodeID", nodeID,
+		"EndPoint", endPoint,
+		"Driver Name", driverName)
+
+	if !instance.DeletionTimestamp.IsZero() {
+		logger.Info("Deleting connection")
+		r.ConnPool.Delete(key)
+
+		if util.ContainsInSlice(instance.Finalizers, csiAddonsNodeFinalizer) {
+			logger.Info("Removing finalizer")
+
+			instance.Finalizers = util.RemoveFromSlice(instance.Finalizers, csiAddonsNodeFinalizer)
```

Comment:
how this CR will be deleted? have we documented creation/deletion of the CSiAddonsNode CR workflow?

Reply:
> how this CR will be deleted?

CSIAddonsNode CR created by sidecar will have ownerRef set to the pod itself, so when the pod is deleted, CR will get deleted too.
https://github.com/csi-addons/kubernetes-csi-addons/blob/main/sidecar/internal/csiaddonsnode/csiaddonsnode.go#L107-L114

> have we documented creation/deletion of the CSiAddonsNode CR workflow?

I guess it's not, https://github.com/csi-addons/kubernetes-csi-addons/blob/main/docs/csiaddonsnode.md
@yati1998 , can update the docs with more details ? 
(possibly when adding deployment guide)